File: clientapp/src/types/TrainingPlan.ts
export type TrainingCategory = number;

export interface TrainingExercise {
    id: string;
    name: string;
    category: TrainingCategory;
    sortOrder: number;
    sets?: number | null;
    reps?: number | null;
    targetWeight?: number | null;
    restSeconds?: number | null;
    durationMin?: number | null;
    distanceKm?: number | null;
    notes?: string | null;
}

export interface TrainingDay {
    id: string;
    name: string;
    sortOrder: number;
    exercises: TrainingExercise[];
}

export interface TrainingPlan {
    id: string;
    name: string;
    isFavorite: boolean;
    createdUtc: string;
    updatedUtc: string;
    code?: string;
    exerciseCount?: number;
    days?: TrainingDay[];
}


export type TrainingPlanUpsert = {
    name: string;
    isFavorite: boolean;
    days: Array<{
        name: string;
        sortOrder: number;
        exercises: Array<{
            name: string;
            category: TrainingCategory;
            sortOrder: number;
            sets?: number | null;
            reps?: number | null;
            targetWeight?: number | null;
            restSeconds?: number | null;
            durationMin?: number | null;
            distanceKm?: number | null;
            notes?: string | null;
        }>;
    }>;
};

File: Gym3000.Api/Entities/TrainingPlan.cs
using System.ComponentModel.DataAnnotations;

namespace Gym3000.Api.Entities;

public class TrainingPlan
{
    public Guid Id { get; set; }

    [Required]
    public string UserId { get; set; } = default!;

    [Required, MaxLength(120)]
    public string Name { get; set; } = default!;

    public bool IsFavorite { get; set; }

    [Required, MaxLength(12)]
    public string Code { get; set; } = default!;

    public DateTime CreatedUtc { get; set; } = DateTime.UtcNow;
    public DateTime UpdatedUtc { get; set; } = DateTime.UtcNow;

    public List<TrainingDay> Days { get; set; } = new();
}
